assets/js/blocks/cart-checkout/cart/full-cart/cart-line-item-row.js

```diff
@@ -23,12 +37,30 @@ const CartLineItemRow = ( { lineItem } ) => {
 	const [ lineQuantity, setLineQuantity ] = useState( quantity );
 	const currency = getCurrency();
 
-	const isDiscounted = subtotal !== total;
-	const fullPrice = isDiscounted ? (
-		<div className="wc-block-cart-item__full-price">
-			<FormattedMonetaryAmount currency={ currency } value={ subtotal } />
-		</div>
-	) : null;
+	const discountAmount = calcPriceDifference( subtotal, total );
+	let fullPrice = null,
+		discountBadge = null;
+	if ( discountAmount > 0 ) {
+		fullPrice = (
+			<div className="wc-block-cart-item__full-price">
+				<FormattedMonetaryAmount
+					currency={ currency }
+					value={ subtotal }
+				/>
+			</div>
+		);
+		discountBadge = (
+			<div className="wc-block-cart-item__discount-badge">
+				<span>
```

Comment:
Could this wrapping `<span>` be removed? We would need to update the stylesheets as well, but I think there is no need to have it here.

Reply:
I _think_ the span is necessary ...

I'm using the (inline) `span` to ensure the "badge" rect is only as big as necessary for the discount. The wrapping `div` is `display: block` so the badge appears on its own line.

<img width="342" alt="Screen Shot 2020-01-14 at 12 02 31 PM" src="https://user-images.githubusercontent.com/4167300/72299462-c46a9d00-36c5-11ea-8b08-11b32c10a7af.png">

Let me know if you have a more elegant suggestion!

Reply:
Oh, I get it now, thank you!

An alternative approach to remove this `<span>` would be:
* Convert `.wc-block-cart-item__line-total` to a `<div>`.
* Add `display: inline-block;` to `.wc-block-cart-item__discount-badge`.

Then you should be able to remove the `<span>` and it will still look the same (if I didn't miss any edge case).

Reply:
I like it - this way is more explicit and consistent. Changed here: 9b5a5a323c1a22b2b19fab8ea18e6351b7dc6c95